agera/src/main/java/com/google/android/agera/Observables.java

```diff
@@ -138,32 +140,45 @@ public static UpdateDispatcher updateDispatcher(
     return new AsyncUpdateDispatcher(activationHandler);
   }
 
-  private static final class CompositeObservable extends BaseObservable implements Updatable {
+  private static final class CompositeObservable implements Observable {
     @NonNull
     private final Observable[] observables;
+    @NonNull
+    private final Map<Updatable, Updatable> updatables;
```

Comment:
List<Updatable> could be a better choice here, because:
* we don't require to pass immutable updatables.
* hashCode() could be very slow